IT fails in anvilprod due to multiple values for is_supplementary

https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/4281

for branch https://github.com/DataBiosphere/azul/tree/issues/hannes-ucsc/5015-anvilprod

and on commit https://github.com/DataBiosphere/azul/commit/4fe493e423aabc5859e4dff9c3019483ef3ec31d

```
======================================================================
ERROR: test_indexing (integration_test.IndexingIntegrationTest) [catalog_complete] (catalog='anvil-it')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 383, in subTest
    yield
  File "/builds/ucsc/azul/test/integration_test.py", line 991, in _assert_catalog_complete
    indexed_fqids = self._get_indexed_bundles(catalog)
  File "/builds/ucsc/azul/test/integration_test.py", line 966, in _get_indexed_bundles
    is_supplementary = one(is_supplementary)
  File "/build/.venv/lib/python3.9/site-packages/more_itertools/more.py", line 554, in one
    raise too_long or ValueError(msg)
ValueError: Expected exactly one item in iterable, but got False, True, and perhaps more.
----------------------------------------------------------------------
Ran 14 tests in 298.301s
FAILED (errors=1, skipped=2)
make: *** [Makefile:237: integration_test] Error 1
Cleaning up project directory and file based variables 00:01
ERROR: Job failed: exit code 1
```

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/test/integration_test.py
+++ b/test/integration_test.py
@@ -957,15 +957,17 @@ class IndexingIntegrationTest(IntegrationTestCase, AlwaysTearDownTestCase):
                     version=bundle['bundleVersion']
                 )
                 if config.is_anvil_enabled(catalog):
-                    for file in hit['files']:
-                        is_supplementary = file['is_supplementary']
-                        if isinstance(is_supplementary, list):
-                            is_supplementary = one(is_supplementary)
-                        if is_supplementary:
-                            bundle_fqid['entity_type'] = BundleEntityType.supplementary.value
-                            break
-                    else:
-                        bundle_fqid['entity_type'] = BundleEntityType.primary.value
+                    # Biosamples are used as bundle entities for primary bundles
+                    # and are never present in supplementary bundles.
+                    # We cannot use the `files.is_supplementary` field to
+                    # determine whether a bundle is supplementary or not due to
+                    # false positives in the current snapshots
+                    # (https://github.com/DataBiosphere/azul/issues/5229)
+                    bundle_fqid['entity_type'] = (
+                        BundleEntityType.primary.value
+                        if hit['biosamples'] else
+                        BundleEntityType.supplementary.value
+                    )
                 bundle_fqid = self.repository_plugin(catalog).resolve_bundle(bundle_fqid)
                 indexed_fqids.add(bundle_fqid)
         return indexed_fqids

